java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/CheckVerifier.java

```diff
@@ -182,36 +181,71 @@ private void assertMultipleIssue(Set<AnalyzerMessage> issues) throws AssertionEr
     }
   }
 
-  private static boolean isLinear(AnalyzerMessage issue) {
+  enum RemediationFunction {
+    LINEAR, CONST
+  }
+
+  static class RuleJSON {
+    static class Remediation {
+      String func;
+    }
+    Remediation remediation;
+  }
+
+  @CheckForNull
+  private static RemediationFunction remediationFunction(AnalyzerMessage issue) {
+    String ruleKey = ruleKey(issue);
+    try {
+      RuleJSON rule = getRuleJSON(ruleKey);
+      if (rule.remediation == null) {
+        return null;
+      }
+      switch (rule.remediation.func) {
+        case "Linear":
+          return RemediationFunction.LINEAR;
+        case "Constant/Issue":
+          return RemediationFunction.CONST;
+        default:
+          return null;
+      }
+    } catch (IOException e) {
+      // Failed to open json file, as this is not part of API yet, we should not fail because of this
+      return null;
```

Comment:
I think it should rethrow with a correct message rather than defaulting to some behavior. Better to fail when something unexpected happens (and test that).